examples/custom_solvers.jl

```diff
@@ -16,19 +16,18 @@ model = model_LDA(lattice, atoms, positions)
 basis = PlaneWaveBasis(model; Ecut=5, kgrid=[1, 1, 1]);
 
 # We define our custom fix-point solver: simply a damped fixed-point
-function my_fp_solver(f, x0, max_iter; tol)
+function my_fp_solver(f, x0, info0; maxiter)
     mixing_factor = .7
     x = x0
-    fx = f(x)
-    for n = 1:max_iter
-        inc = fx - x
-        if norm(inc) < tol
+    info = info0
+    for n = 1:maxiter
+        fx, info = f(x, info)
+        if info.converged
             break
         end
-        x = x + mixing_factor * inc
-        fx = f(x)
+        x = x + mixing_factor * (fx - x)
     end
-    (; fixpoint=x, converged=norm(fx-x) < tol)
+    (; fixpoint=x, info)
 end;
```

Comment:
In line with the first example here I would probably show the "proper way" to supply new solvers, but also show with a custom `is_converged` callback how to customise the convergence check.

Reply:
I added a short note on this one the custom solvers docs, and appended a customizing example with stresses in analogy to ScfConvergenceForce in [3a37b2a](https://github.com/JuliaMolSim/DFTK.jl/pull/811/commits/3a37b2a99f81d5be3151d7c88d96033777857c9a). Though I wonder whether this is too complicated.